youtube_dl/extractor/tumblr.py

```diff
@@ -97,6 +105,31 @@ class TumblrIE(InfoExtractor):
         'add_ie': ['Instagram'],
     }]
 
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        if username is None:
+            return
+        self.report_login()
+        webpage = self._download_webpage(self._LOGIN_URL, None, False)
+        form = self._hidden_inputs(webpage)
+        form.update({
+            'user[email]': username,
+            'user[password]': password
+        })
+        post_data = urlencode_postdata(form)
+        login_request = sanitized_Request(self._LOGIN_URL, post_data)
+        login_request.add_header('Content-Type', 'application/x-www-form-urlencoded')
+        login_request.add_header('Referer', self._LOGIN_URL)
+        login_response = self._download_webpage(login_request, None, False, 'Wrong login info')
+
+        # Check the login response from Tumblr for an error message and fail the extraction if we find one.
+        login_errors = self._search_regex(r'Tumblr\.RegistrationForm\.errors = \[(.*)\]', login_response, 'login errors', False, False)
```

Comment:
1. Relax whitespace.
2. default and fatal are not used together.
3. Should not match empty string.

Reply:
- Could you please elaborate on what you mean by "relax whitespace?" Are you saying I should modify the RegEx to ignore whitespace?
- Please correct me if I'm wrong, but I don't think default and fatal are being used together here. The default is set to false and the fatal flag (which is true by default) is set to false. If I shouldn't be setting both options what would you suggest I do instead?

Reply:
1. Regex should match any variation of whitespace around `=`.
2. Once you provide default, fatal is not used thus it's pointless to provide fatal along with default.